api/base/filters.py

```diff
@@ -537,4 +538,8 @@ def preprints_queryset(self, base_queryset, auth_user, allow_contribs=True):
         if not moderator_for.exists():
             base_queryset = base_queryset.exclude(date_withdrawn__isnull=False, ever_public=False)
 
-        return base_queryset.annotate(default=Exists(sub_qs)).filter(Q(default=True) & query).distinct('id', 'created')
+        ret = base_queryset.annotate(default=Exists(sub_qs)).filter(Q(default=True) & query)
+        # The auth subquery currently results in duplicates returned
+        # https://openscience.atlassian.net/browse/OSF-9058
+        # TODO: Remove need for .distinct using correct subqueries
+        return ret.distinct('id', 'created') if include_non_public else ret
```

Comment:
This change imposes the limitation that only public preprints will be returned when metrics are requested.